src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WcfService.cs

```diff
@@ -251,6 +251,24 @@ public string GetRequestCustomHeader(string customHeaderName, string customHeade
             return infos;
         }
 
+        public string GetIncomingMessageHeadersMessage(string customHeaderName, string customHeaderNS)
+        {
+            try {
+                MessageHeaders headers = OperationContext.Current.IncomingMessageHeaders;
+                TestMessageHeader header = headers.GetHeader<TestMessageHeader>(customHeaderName, customHeaderNS);
+                if (header != null)
+                {
+                    return header.Message;
+                }
+            }
+            catch
+            {
+
```

Comment:
I recommend we don't catch and ignore all exceptions in this operation.  Better to let any exception propagate back to the client.